src/main/java/net/neoforged/neoforge/registries/DeferredRegister.java

```diff
@@ -5,319 +5,267 @@
 
 package net.neoforged.neoforge.registries;
 
-import com.google.common.collect.Multimaps;
-import com.google.common.collect.SetMultimap;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashSet;
-import java.util.IdentityHashMap;
+import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Objects;
 import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
 import java.util.function.Supplier;
+import net.minecraft.core.Holder;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.core.registries.Registries;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.tags.TagKey;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.neoforged.bus.api.IEventBus;
-import net.neoforged.bus.api.SubscribeEvent;
-import net.neoforged.neoforge.registries.tags.ITagManager;
-import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
- * Utility class to help with managing registry entries.
- * Maintains a list of all suppliers for entries and registers them during the proper Register event.
+ * A DeferredRegister is a helper class to aid in registering objects to modded and {@linkplain BuiltInRegistries vanilla registries} and provide deferred suppliers to access those objects. <br>
+ * This class maintains a list of all suppliers for entries and registers them during the proper {@link RegisterEvent} event, after being {@linkplain #register(IEventBus) registered} to an event bus.
  * Suppliers should return NEW instances every time.
- *
+ * <p>
  * Example Usage:
- * 
+ *
  * <pre>{@code
- * private static final DeferredRegister<Item> ITEMS = DeferredRegister.create(ForgeRegistries.ITEMS, MODID);
- * private static final DeferredRegister<Block> BLOCKS = DeferredRegister.create(ForgeRegistries.BLOCKS, MODID);
+ * private static final DeferredRegister.Items ITEMS = DeferredRegister.createItems(MODID);
+ * private static final DeferredRegister.Blocks BLOCKS = DeferredRegister.createBlocks(MODID);
```

Comment:
There should be an example for the generic `DeferredRegister` as well, and not just for the specific cases of `Block`s and `Item`s.

Reply:
Fixed in f2801cb37.